User permissions

ltrilety opened the pull request:
Test for user roles permissions

ltrilety committed:
Add test for read-only user permissions

ltrilety committed:
Add test for power user permissions

ltrilety committed:
Add test for admin user permissions

ltrilety committed:
Fix indentation

mbukatov commented on web/tendrl/admin_user_permissions.rst:
```diff
@@ -0,0 +1,131 @@
+Multiuser - Positive login
+**************************
```
The headline (and the name of the test case) seems to be wrong.

mbukatov commented on web/tendrl/admin_user_permissions.rst:
```diff
@@ -0,0 +1,131 @@
+    :result:
+        Filter and order fields should be available and working.
+        There should be functional link to Grafana for imported clusters.
+        It should be possible to import, manage, unmanage and forget cluster.
```
Why do you think that Tendrl will provide unmanage or forget cluster functionality? As far as I know this feature would not be directly available in the ui, at least for the upcoming upstream release we are targeting in these test cases. I would suggest to either link to the specification/issue/ux desing for this feature or to remove this from the test case. Moreover it would be nice to link to the definition of what manage, unmanage and forget actually mean in this context (without that, this explanation is not very useful wrt to expected functionality).

mbukatov commented on web/tendrl/admin_user_permissions.rst:
```diff
@@ -0,0 +1,131 @@
+        Choose some cluster
+    :result:
+        There should be cluster Hosts list, Bricks list, Volumes list,
+          Configuration, Events list and Tasks list available.
```
syntax note: different indentation of this line (wrt to the previous one) will make this particular block of text presented as a definition, it would be better to use the same indentation for both lines

mbukatov commented on web/tendrl/normal_user_permissions.rst:
```diff
@@ -0,0 +1,122 @@
+Multiuser - Positive login
+**************************
```
Wrong name of the test case.

mbukatov commented on web/tendrl/admin_user_permissions.rst:
```diff
@@ -0,0 +1,131 @@
+
+**UNDER CONSTRUCTION - NEED UPDATE**
+
+Login as user with 'Admin' role.
```
Shouldn't this login line be in setup section?

mbukatov commented on web/tendrl/normal_user_permissions.rst:
```diff
@@ -0,0 +1,122 @@
+
+**UNDER CONSTRUCTION - NEED UPDATE**
+
+Login as user with 'Power User' ('Normal User') role.
```
Shouldn't this login line be in setup section?

mbukatov commented on web/tendrl/readonly_user_permissions.rst:
```diff
@@ -0,0 +1,122 @@
+Multiuser - Positive login
+**************************
```
wrong name

mbukatov commented on web/tendrl/readonly_user_permissions.rst:
```diff
@@ -0,0 +1,122 @@
+
+**UNDER CONSTRUCTION - NEED UPDATE**
+
+Login as user with 'Read-Only' ('limited') role.
```
Shouldn't this login line be in setup section?

mbukatov commented on web/tendrl/readonly_user_permissions.rst:
```diff
@@ -0,0 +1,122 @@
+        Choose some cluster
+    :result:
+        There should be cluster Hosts list, Bricks list, Volumes list,
+          Configuration, Events list and Tasks list available.
```
wrong indentation

mbukatov commented on web/tendrl/normal_user_permissions.rst:
```diff
@@ -0,0 +1,122 @@
+        Choose some cluster
+    :result:
+        There should be cluster Hosts list, Bricks list, Volumes list,
+          Configuration, Events list and Tasks list available.
```
wrong indentation

mbukatov commented on web/tendrl/normal_user_permissions.rst:
```diff
@@ -0,0 +1,122 @@
+    :result:
+        Filter and order fields should be available and working.
+        It should be possible to open task details.
+
```
Should we also test that one can't add new or edit/remove old user here?

mbukatov reviewed (changes requested):
Ok, let's start with this kind of description for now.

See my questions and suggestions below.

ltrilety committed:
Fix issues

ltrilety committed:
rename normal user test

mkudlej reviewed (commented):
LGTM

mbukatov committed:
drop forgotten line from description

mbukatov committed:
minor rst syntax fix

Empty line between text and item list is necessary, otherwise items would
not be recongined as an item list.

mbukatov committed:
drop note about cluster forget/unmanage

This feature is not specified anywhere, and so it will not be part of
upcoming release we are trying to test.

mbukatov committed:
drop extraneous line

mbukatov committed:
drop note about cluster forget/unmanage

mbukatov committed:
drop extraneous line

mbukatov committed:
rst syntax fix

mbukatov committed:
rst syntax fix
